src/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -0,0 +1,560 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    internal sealed class RsaPaddingProcessor
+    {
+        private static readonly byte[] s_eightZeros = new byte[8];
+
+        private static readonly Dictionary<HashAlgorithmName, RsaPaddingProcessor> s_lookup =
+            new Dictionary<HashAlgorithmName, RsaPaddingProcessor>();
+
+        private readonly HashAlgorithmName _hashAlgorithmName;
+        private readonly int _hLen;
+
+        private RsaPaddingProcessor(HashAlgorithmName hashAlgorithmName, int hLen)
+        {
+            _hashAlgorithmName = hashAlgorithmName;
+            _hLen = hLen;
+        }
+
+        internal int HashLength => _hLen;
+
+        internal static RsaPaddingProcessor OpenProcessor(HashAlgorithmName hashAlgorithmName)
+        {
+            lock (s_lookup)
```

Comment:
Same ConcurrentDictionary question.  It'd be nice to avoid a global lock like this for already initialized values.

Reply:
Changed both to ConcurrentDictionary.